chore: version to 0.42.0 (#3259)
prefix-dev/pixi: CHANGELOG.md

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

### [0.42.0] - 2025-03-04
#### ✨ Highlights

> This release introduces a new way of [dealing with Virtual Packages](https://github.com/prefix-dev/pixi/pull/2849).

Reviewer: maybe something like this:
```suggestion
This release introduces an improved way of [dealing with Virtual Packages](https://github.com/prefix-dev/pixi/pull/2849).
```
